java/ql/test/utils/model-generator/typebasedflow/p/Stream.java

```diff
@@ -0,0 +1,249 @@
+        return null;
+    }
+
+    // Collector is not a functional interface, so this is not supported
+    public <R, A> R collect(Collector<? super T, A, R> collector) {
+        return null;
+    }
+
+    // MaD=p;Stream;true;concat;(Stream,Stream);;Argument[0].Element;ReturnValue.Element;value;generated
+    // MaD=p;Stream;true;concat;(Stream,Stream);;Argument[1].Element;ReturnValue.Element;value;generated
+    public static <T> Stream<T> concat(Stream<? extends T> a, Stream<? extends T> b) {
+        return null;
+    }
+
+    // MaD=p;Stream;true;distinct;();;Argument[-1].Element;ReturnValue.Element;value;generated
+    public Stream<T> distinct() {
+        return null;
+    }
+
+    public static <T> Stream<T> empty() {
+        return null;
+    }
+
+    // MaD=p;Stream;true;filter;(Predicate);;Argument[-1].Element;Argument[0].Parameter[0];value;generated
+    // MaD=p;Stream;true;filter;(Predicate);;Argument[-1].Element;ReturnValue.Element;value;generated
+    public Stream<T> filter(Predicate<? super T> predicate) {
+        return null;
+    }
+
+    // MaD=p;Stream;true;findAny;();;Argument[-1].Element;ReturnValue.SyntheticField[ArgType0];value;generated
+    public Optional<T> findAny() {
+        return null;
+    }
+
+    // MaD=p;Stream;true;findFirst;();;Argument[-1].Element;ReturnValue.SyntheticField[ArgType0];value;generated
+    public Optional<T> findFirst() {
+        return null;
+    }
+
+    // public <R> Stream<R> flatMap(Function<? super T, ? extends Stream<? extends
+    // R>> mapper) {
+    // return null;
+    // }
+
+    // MaD=p;Stream;true;flatMapToDouble;(Function);;Argument[-1].Element;Argument[0].Parameter[0];value;generated
+    public DoubleStream flatMapToDouble(Function<? super T, ? extends DoubleStream> mapper) {
+        return null;
+    }
+
+    // MaD=p;Stream;true;flatMapToInt;(Function);;Argument[-1].Element;Argument[0].Parameter[0];value;generated
+    public IntStream flatMapToInt(Function<? super T, ? extends IntStream> mapper) {
+        return null;
+    }
+
+    // MaD=p;Stream;true;flatMapToLong;(Function);;Argument[-1].Element;Argument[0].Parameter[0];value;generated
+    public LongStream flatMapToLong(Function<? super T, ? extends LongStream> mapper) {
+        return null;
+    }
+
+    // MaD=p;Stream;true;forEach;(Consumer);;Argument[-1].Element;Argument[0].Parameter[0];value;generated
+    public void forEach(Consumer<? super T> action) {
+    }
+
+    // MaD=p;Stream;true;forEachOrdered;(Consumer);;Argument[-1].Element;Argument[0].Parameter[0];value;generated
+    public void forEachOrdered(Consumer<? super T> action) {
+    }
+
+    // MaD=p;Stream;true;generate;(Supplier);;Argument[0].ReturnValue;ReturnValue.Element;value;generated
+    public static <T> Stream<T> generate(Supplier<T> s) {
+        return null;
+    }
+
+    // Model generator adds a couple of extra models, which can't be
```

Comment:
All four of these models look correct to me?

Reply:
Yes, you are right! I will remove the comment (I had overlooked that the "seed" is the first element of the sequence, which is why I thought that Argument[0] -> ReturnValue.Element was incorrect - but that is not the case).